src/libmongoc/src/mongoc/mongoc-server-monitor.c

```diff
@@ -1141,10 +1145,14 @@ static BSON_THREAD_FUN (_server_monitor_rtt_thread, server_monitor_void)
       }
       bson_mutex_unlock (&server_monitor->shared.mutex);
 
-      _server_monitor_ping_server (server_monitor, &rtt_ms);
-      if (rtt_ms != MONGOC_RTT_UNSET) {
-         mongoc_server_description_t *sd;
+      sd = mongoc_topology_description_server_by_id (
+         &server_monitor->topology->description,
+         server_monitor->description->id,
+         &error);
+      hello_ok = sd ? sd->hello_ok : false;
```

Comment:
Ah, thanks for pointing this out. I think this could happen, but as with many things Versioned API, we have to assume that our users will only declare an Api Version when running clusters only consisting of MongoDB 5.0+ with no downgrades happening. In the off-chance that someone is using Versioned API and still downgrades a server in the cluster, they'll have to deal with this like they would if they were using other features (e.g. using commands that are only supported in newer server versions).

Agree with leaving this as is. 👍

Reply:
> we have to assume that our users will only declare an Api Version when running clusters only consisting of MongoDB 5.0+ with no downgrades happening

That seems like a sensible assumption.